SourceX/controls/keymapper.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <string>
+#include <functional>
+#include <memory>
+#include <vector>
+#include <unordered_map>
+
+namespace dvl {
+
+// The Keymapper maps keys to actions.
+class Keymapper final
+{
+public:
+	using SetKeyFunction = std::function<void(const std::string &key, const std::string &value)>;
+	using GetKeyFunction = std::function<std::string(const std::string &key)>;
+
+	// Action represents an action that can be triggered using a keyboard shortcut.
+	class Action final
+	{
+	public:
+		// Can this action be triggered while the player is dead?
+		enum class IfDead
+		{
+			Allow,
+			Ignore,
+		};
+
+		Action(const std::string &name, int defaultKey, std::function<void()> action):
+			name{name}, defaultKey{defaultKey}, action{action}, ifDead{IfDead::Ignore} {}
+		Action(const std::string &name, int defaultKey, std::function<void()> action, IfDead ifDead):
```

Comment:
`std::function` means the action is copied.
Since all the actions are defined beforehand, perhaps this should be a (C-style) function pointer (as we don't have `std::function_ref` yet)?